integration/spark-datasource/src/main/scala/org/apache/spark/sql/carbondata/execution/datasources/CarbonFileIndexReplaceRule.scala

```diff
@@ -82,4 +82,23 @@ class CarbonFileIndexReplaceRule extends Rule[LogicalPlan] {
       fileIndex
     }
   }
+
+  /**
+   * Get datafolders recursively
+   */
+  private def getDataFolders(carbonFile: CarbonFile): Seq[CarbonFile] = {
+    val files = carbonFile.listFiles()
+    var folders: Seq[CarbonFile] = Seq()
+    files.foreach { f =>
+      if (f.isDirectory) {
+        val files = f.listFiles()
+        if (files.nonEmpty && !files(0).isDirectory) {
+          folders = Seq(f) ++ folders
+        } else {
+          folders = getDataFolders(f) ++ folders
```

Comment:
This statement can be moved under files.nonEmpty check

Reply:
In the next call to getDataFolders() for empty folder it skips the iteration. So, no problem with the logic. you can ignore the comment.